File: articles/myth-symbol-and-the-archetypal-imagination/fortran/motif_weight.f90
! Myth, Symbol, and the Archetypal Imagination
! Fortran example: simple motif co-occurrence weight

program motif_weight_demo
  implicit none

  integer :: weight
  real :: context_count
  real :: motif_total
  real :: share

  if (.true. .and. .true.) then
    weight = 1
  else
    weight = 0
  end if

  context_count = 4.0
  motif_total = 10.0

  if (motif_total > 0.0) then
    share = context_count / motif_total
  else
    share = 0.0
  end if

  print *, "Synthetic motif co-occurrence weight:", weight
  print *, "Synthetic context share:", share
end program motif_weight_demo

File: articles/myth-symbol-and-the-archetypal-imagination/go/motif_weight.go
package main

import "fmt"

func cooccurrenceWeight(motifAPresent bool, motifBPresent bool) int {
	if motifAPresent && motifBPresent {
		return 1
	}
	return 0
}

func contextShare(contextCount float64, motifTotal float64) float64 {
	if motifTotal <= 0 {
		return 0
	}
	return contextCount / motifTotal
}

func main() {
	weight := cooccurrenceWeight(true, true)
	share := contextShare(4.0, 10.0)

	fmt.Printf("Synthetic motif co-occurrence weight: %d\n", weight)
	fmt.Printf("Synthetic context share: %.3f\n", share)
}
